src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -3806,12 +3866,13 @@ subroutine ckgcov_a_en_new_factorization(ig,z,a_en)
 ! Passed Variables
   integer(i_kind),intent(in   ) :: ig
   type(gsi_bundle),intent(inout) :: a_en(n_ens)
-  real(r_kind),dimension(nval_lenz_en),intent(in   ) :: z
+  real(r_kind),dimension(nval_loc_en),intent(in   ) :: z
 
 ! Local Variables
-  integer(i_kind) ii,k,iadvance,iback,is,ie,ipnt,istatus
+  integer(i_kind) ii,i,j,k,iadvance,iback,is,ie,ipnt,istatus
   real(r_kind) hwork(grd_loc%nlat*grd_loc%nlon*(grd_loc%kend_alloc-grd_loc%kbegin_loc+1))
-!NOTE:   nval_lenz_en = nhoriz*(grd_loc%kend_alloc-grd_loc%kbegin_loc+1)
+!NOTE: Except for MGBF-based localization,
```

Comment:
@shoyokota the new comments are confusing. . I 'd suggest that ,  add " when L-mbgf , nval_loc_en is used as the size of z and z2 ...."

Reply:
Now I modified this comment not to cause confusion.